crates/ruff_linter/src/rules/flake8_implicit_str_concat/rules/explicit.rs

```diff
@@ -35,45 +39,133 @@ use crate::settings::LinterSettings;
 pub(crate) struct ExplicitStringConcatenation;
 
 impl Violation for ExplicitStringConcatenation {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Always;
+
     #[derive_message_formats]
     fn message(&self) -> String {
         "Explicitly concatenated string should be implicitly concatenated".to_string()
     }
+
+    fn fix_title(&self) -> Option<String> {
+        Some("Remove redundant '+' operator to implicitly concatenate".to_string())
+    }
 }
 
 /// ISC003
-pub(crate) fn explicit(
-    expr: &Expr,
-    locator: &Locator,
-    settings: &LinterSettings,
-) -> Option<Diagnostic> {
+pub(crate) fn explicit(expr: &Expr, checker: &Checker) -> Option<Diagnostic> {
     // If the user sets `allow-multiline` to `false`, then we should allow explicitly concatenated
     // strings that span multiple lines even if this rule is enabled. Otherwise, there's no way
     // for the user to write multiline strings, and that setting is "more explicit" than this rule
     // being enabled.
-    if !settings.flake8_implicit_str_concat.allow_multiline {
+    if !checker.settings.flake8_implicit_str_concat.allow_multiline {
         return None;
     }
 
-    if let Expr::BinOp(ast::ExprBinOp {
-        left,
-        op,
-        right,
-        range,
-    }) = expr
-    {
+    if let Expr::BinOp(bin_op) = expr {
+        let ast::ExprBinOp {
+            left,
+            right,
+            range,
+            op,
```

Comment:
Unrelated to your PR but
```suggestion
            op: Operator::Add,
```

and you can then remove the match on line 71

We could also inline the match on line 72 and 77 by rewriting this to

```
left @ (Expr::FString(_) | Expr::StringLiteral(_) | Expr::BytesLiteral(_)
right @ (...)
```

and the same for right

Reply:
I don't think the inlining of `left` and `right` works because they are `Box<Expr>` but the `box` destructuring is experimental, so we cannot do. 

```rust 
if let ast::ExprBinOp {
    left: left @ box (Expr::FString(_) | Expr::StringLiteral(_) | Expr::BytesLiteral(_)),
    right: right @ box (Expr::FString(_) | Expr::StringLiteral(_) | Expr::BytesLiteral(_)),
    range,
    op: Operator::Add,
} = bin_op {...}
```